pep-0716.rst

```diff
@@ -0,0 +1,319 @@
+
+The ``project.name`` field **MUST** be normalized during comparison.
+
+
+.dist-info directories
+----------------------
+
+The directory name follows the pattern of ``{name}-{version}.dist-info``.
+
+The ``name`` field **MUST** be normalized, with any resulting ``-`` escaped to ``_``.
+
+Tools that read an arbitrary ``.dist-info`` directory **MUST** be prepared to
+accept unnormalized values, however tools that work only on *new* ``.dist-info``
+directories **SHOULD** validate that all values are normalized.
+
+
+Source and Binary Distributions
+-------------------------------
+
+Both the sdist and bdist specifications incorporate the project name in their
+filenames (``{name}-{version}.tar.gz`` and
+``{distribution}-{version}(-{build tag})?-{python tag}-{abi tag}-{platform tag}.whl``
+respectively).
+
+The ``name`` field **MUST** be non-normalized, with the exception that any ``-``
+**MUST** be escaped to be ``_``.
+
+Tools that accept an arbitrary distribution **MUST** be prepared to accept both
+non-normalized and normalized filenames. However, tools that only work on *new*
+distributions **SHOULD** validate that the distribution filenames are not
+normalizing ``name``.
+
+
+Simple Repository API
+---------------------
+
+The project name, when returned in the "index" URL (e.g. ``/simple/``)
+**MUST** be non-normalized.
+
+The project name when used in the URL (e.g. ``/simple/$project/``) **MUST** be
+normalized.
+
+The project name, when used on the Project detail page
+(e.g. ``/simple/$project/``), **MUST** be non-normalized.
+
+Tools that read values for filenames and names from the Simple Repository API
+**MUST** be prepared to handle both normalized and non-normalized names.
+
+
+Backwards Compatibility
+=======================
+
+This PEP breaks compatibility in a few ways:
+
+* Tools that are currently emitting filenames where ``name`` has been normalized
+  in accordance with the current spec are immediately no longer compliant and
+  must be updated to emit non-normalized names.
+
+  * This is mitigated by the fact that all tools are required to continue to
+    accept both normalized and non-normalized filenames unless they *know* that
+    they only work on *new* distributions (PyPI uploads, ``pyproject-build``, etc).
+
+* Tools that emit normalized names into ``METADATA``, ``PKG-INFO``, or when
+  uploading to a repository are immediately no longer compliant and must be
+  updated to emit non-normalized names.
+
+  * It's unclear in the current spec whether names were intended to be normalized
+    in this case or not, but the practice of normalization here has caused a
+    number of people to be confused why their names are different from what
+    they've entered.
+
+* Tools that are currently emitting the names in the simple API (outside of the URL
+  itself) as normalized, which is either allowed or required by the spec
+  currently are immediately not longer complaint and must be updated to emit
+  non-normalized names.
```

Comment:
The PEP should probably list which tools this is known to affect. After reading the PEP to here, I'm sufficiently overwhelmed as to be unsure whether this is referring to PyPI or Artifactory, or some other index software.

Reply:
My impression here was that this specific bit referred to neither, and rather referred to installers using the normalized rather than the unnormalized name in user-facing UI text—but clearly, given there is evident confusion here it should be clarified appropriately.